File: core/runtime_language/wwx_interpreter.py
from __future__ import annotations

from typing import Any, Dict

from core.runtime_language.wwx_compiler import compile_wwx
from core.runtime_language.wwx_parser import parse_wwx


def interpret_wwx(source: str, tick: int = 0) -> Dict[str, Any]:
    parsed = parse_wwx(source)
    compiled = compile_wwx(parsed)
    results = []
    for step in compiled.get("plan", {}).get("steps", []):
        results.append({
            "action": step["action"],
            "target": step["target"],
            "simulated": step["action"] in ("extract", "sync", "replay"),
            "tick": tick,
            "executed": step["action"] == "execute",
        })
    return {
        "parsed": parsed,
        "compiled": compiled,
        "results": results,
        "deterministic": True,
        "bounded": True,
    }

File: src/runtime_language/wwxInterpreter.ts
/**
 * Converted from Python: core/runtime_language/wwx_interpreter.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";
import { compileWwx } from "./wwxCompiler.js";
import { parseWwx } from "./wwxParser.js";

export function interpretWwx(source: any, tick: any = 0): any {
  var parsed: any = parseWwx(source);
  var compiled: any = compileWwx(parsed);
  var results: any[] = [];
  var step: any;
  for (step of py.iter(py.get(py.get(compiled, "plan", {}), "steps", []))) {
    py.listAppend(results, {"action": py.at(step, "action"), "target": py.at(step, "target"), "simulated": py.contains(["extract", "sync", "replay"], py.at(step, "action")), "tick": tick, "executed": py.eq(py.at(step, "action"), "execute")});
  }
  return {"parsed": parsed, "compiled": compiled, "results": results, "deterministic": true, "bounded": true};
}
export { compileWwx, parseWwx };
